test/odbc/psqlodbc/psqlodbc_tests_common.cpp

```diff
@@ -0,0 +1,598 @@
+
+    rcode = SQLColAttribute(odbcHandler.GetStatementHandle(),
+                            i,
+                            SQL_DESC_CASE_SENSITIVE, // Get the scale of the column
+                            NULL,
+                            0,
+                            NULL,
+                            (SQLLEN*) &isCaseSensitive); 
+    EXPECT_EQ(rcode, SQL_SUCCESS);
+    EXPECT_EQ(isCaseSensitive, caseSensitivityExpected[i-1]);
+
+    rcode = SQLColAttribute(odbcHandler.GetStatementHandle(),
+                            i,
+                            SQL_DESC_LITERAL_PREFIX, // Get the prefix of the column
+                            name,
+                            BUFFER_SIZE,
+                            NULL,
+                            NULL); 
+    EXPECT_EQ(rcode, SQL_SUCCESS);
+    EXPECT_EQ(string(name), prefixExpected[i-1]);
+
+    rcode = SQLColAttribute(odbcHandler.GetStatementHandle(),
+                            i,
+                            SQL_DESC_LITERAL_SUFFIX, // Get the suffix character of the column
+                            name,
+                            BUFFER_SIZE,
+                            NULL,
+                            NULL);
+    EXPECT_EQ(rcode, SQL_SUCCESS); 
+    EXPECT_EQ(string(name), suffixExpected[i-1]);
+  }
+
+  rcode = SQLFetch(odbcHandler.GetStatementHandle());
+  EXPECT_EQ(rcode, SQL_NO_DATA);
+}
+
+void testTableCreationFailure(ServerType serverType, const string &tableName, const vector<vector<pair<string, string>>> &invalidColumns) {
+  OdbcHandler odbcHandler(Drivers::GetDriver(serverType));
+  RETCODE rcode;
+
+  odbcHandler.Connect();
+  odbcHandler.AllocateStmtHandle();
+
+  for (int i = 0; i < invalidColumns.size(); i++) {
+    rcode = SQLExecDirect(odbcHandler.GetStatementHandle(),
+                        (SQLCHAR*) CreateTableStatement(tableName, invalidColumns[i]).c_str(),
+                        SQL_NTS);
+
+    EXPECT_EQ(rcode, SQL_ERROR);
+  }
+}
+
+void testInsertionSuccess(ServerType serverType, const string &tableName, const string &orderByColumnName, 
+  const vector<string> &insertedValues, const vector<string> &expectedInsertedValues) {
+
+  insertValuesInTable(serverType, tableName, insertedValues, false);
+  verifyValuesInObject(serverType, tableName, orderByColumnName, insertedValues, expectedInsertedValues);
+}
+
+void testInsertionFailure(ServerType serverType, const string &tableName, const string &orderByColumnName, 
+  const vector<string> &invalidInsertedValues, bool isNumericInsert) {
+
+  OdbcHandler odbcHandler(Drivers::GetDriver(serverType));
+  odbcHandler.Connect(true);
+
+  RETCODE rcode;
+
+   for (int i = 0; i < invalidInsertedValues.size(); i++) {
+    string insertString{};
+
+    if (isNumericInsert) {
+      insertString = "(" + std::to_string(i) + "," + invalidInsertedValues[i] + ")";
+    }
+    else {
+      insertString = "(" + std::to_string(i) + "," + "'" + invalidInsertedValues[i] + "'" + ")";
```

Comment:
```suggestion
      insertString = "(" + std::to_string(i) + ",\'" + invalidInsertedValues[i] + "\')";
```
Suggestion is more about having less concatenation with strings.
I know Java doesn't like it all that much due to Strings being immutable, unsure how much of a performance hit C++ would be.

Reply:
Valid concern. Fixed here and in other relevant areas.